src/main/drivers/bus_spi_ll.c

```diff
@@ -216,18 +218,73 @@ bool spiTransfer(SPI_TypeDef *instance, const uint8_t *txData, uint8_t *rxData,
     return true;
 }
 
-void spiSetDivisor(SPI_TypeDef *instance, uint16_t divisor)
+static uint16_t spiDivisorToBRbits(SPI_TypeDef *instance, uint16_t divisor)
 {
 #if !(defined(STM32F1) || defined(STM32F3))
     // SPI2 and SPI3 are on APB1/AHB1 which PCLK is half that of APB2/AHB2.
 
     if (instance == SPI2 || instance == SPI3) {
         divisor /= 2; // Safe for divisor == 0 or 1
     }
+#else
+    UNUSED(instance);
 #endif
 
+    return divisor ? ((ffs(divisor | 0x100) - 2) << SPI_CR1_BR_Pos) : 0;
+}
+
+void spiSetDivisor(SPI_TypeDef *instance, uint16_t divisor)
+{
     LL_SPI_Disable(instance);
-    LL_SPI_SetBaudRatePrescaler(instance, divisor ? (ffs(divisor | 0x100) - 2) << SPI_CR1_BR_Pos : 0);
+    LL_SPI_SetBaudRatePrescaler(instance, spiDivisorToBRbits(instance, divisor));
     LL_SPI_Enable(instance);
 }
+
+#ifdef USE_SPI_TRANSACTION
+void spiBusTransactionInit(busDevice_t *bus, SPIMode_e mode, SPIClockDivider_e divisor)
+{
+    switch (mode) {
+    case SPI_MODE0:
```

Comment:
here, SPI_MODE_POL_LOW_1EDGE would be better.

similar for below.

Reply:
Mode descriptions as comments will be added in the next branch update.

Reply:
no reason not to include the MODE number AND the details.  In my experience I have found a lot of devices that refer to the legacy "modes" and also a lot that do not.  Having the correlation in the name of the mode just makes it easier all round.